lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -177,50 +163,427 @@ def stage_input(self, params):
+                os.makedirs(temp_dir)
+            ret_file = os.path.join(temp_dir, os.path.basename(input_files[0]))
+
+            # take in Genbank file and remove all empty lines from it.
+            with open(ret_file, 'w', buffering=2 ** 20) as f_out:
+                for input_file in input_files:
+                    with open(input_file, 'r') as f_in:
+                        for line in f_in:
+                            line = line.rstrip('\r\n')
+                            if line.strip():
+                                f_out.write(line + '\n')
+            return ret_file
+
+    def _get_pubs(self, record):
+        pub_list = []
+        for in_pub in record.annotations.get('references', []):
+            out_pub = [
+                0,  # pmid
+                "",  # source
+                in_pub.title,
+                "",  # web address
+                "",  # date
+                in_pub.authors,
+                in_pub.journal,
+            ]
+            date_match = re.match("\((\d{4})\)", in_pub.journal)
+            if date_match:
+                out_pub[4] = date_match.group(1)
+            if in_pub.pubmed_id:
+                out_pub[0:4] = [
+                    int(in_pub.pubmed_id),
+                    "PubMed",
+                    in_pub.title,
+                    "http://www.ncbi.nlm.nih.gov/pubmed/{}".format(
+                        in_pub.pubmed_id)]
+            pub_list.append(tuple(out_pub))
+        self.log("Parsed {} publication records".format(len(pub_list)))
+        return set(pub_list)
+
+    def _get_ontology(self, feature):
+        ontology = defaultdict(dict)
+        for key in ("GO_process", "GO_function", "GO_component"):
+            if key in feature.qualifiers:
+                sp = feature.qualifiers[key][0][3:].split(" - ")
+                ontology['GO'][sp[0]] = [1]
+                self.ontologies_present['GO'][sp[0]] = sp[1]
+        # TODO: Support other ontologies
+        return dict(ontology)
+
+    def _parse_features(self, record, source):
+        def _get_id(feat, tags=None):
+            """Assign a id to a feature based on the first tag that exists"""
+            _id = ""
+            if not tags:
+                tags = ['locus_tag', 'gene']
+            while tags and not _id:
+                _id = feat.qualifiers.get(tags.pop(0), [""])[0]
+            return _id
+
+        def _location(feat):
+            strand_trans = ("", "+", "-")
+            loc = []
+            for part in feat.location.parts:
+                if part.strand >= 0:
+                    begin = int(part.start)
+                else:
+                    begin = int(part.end)
+                loc.append((
+                        record.id,
+                        begin + 1,
+                        strand_trans[part.strand],
+                        len(part)))
+            return loc
+
+        def _aliases(feat):
+            keys = ('locus_tag', 'old_locus_tag', 'protein_id',
```

Comment:
We "EC_number" added here as well.  Kind of tricky. It is more of a db_xref, but we want it to go back to its own key value pair in the GB download ex: /EC_number="1.6.99.3"

Reply:
Not Addressed